tests/macros/utils.js

```diff
@@ -63,15 +65,37 @@ function createMacroTestObject(macroName) {
         url: 'https://developer.mozilla.org/'
     };
     let environment = new Environment(pageContext, templates, true);
+    let ctx = environment.prototypeEnvironment;
+
+    /** @type {Map<string, string>} */
+    const macroResults = new Map();
+    const realTemplate = ctx.template;
+    ctx.template = jest.fn(async (name, ...args) => {
+        let macroName = String(name)
+            .replace(/:/g, '-')
+            .toLowerCase();
+        return macroResults.get(macroName) || realTemplate(name, ...args);
+    });
 
     return {
         /**
          * Give the test-case writer access to the macro's globals (ctx).
          * For example, "macro.ctx.env.locale" can be manipulated to something
          * other than 'en-US' or "macro.ctx.wiki.getPage" can be mocked
-         * using "sinon.stub()" to avoid network calls.
+         * using `jest.fn()` to avoid network calls.
+         */
+        ctx,
+
+        /**
+         * A much easier way to handle overwriting the result of the `template`
+         * function for specific macro calls.
+         *
+         * @param {string} name
+         * @param {string} result
          */
-        ctx: environment.prototypeEnvironment,
+        mockTemplate(name, result) {
+            macroResults.set(name.replace(/:/g, '-').toLowerCase(), result);
```

Comment:
Optionally, you could add code here so that if result is null, then instead of setting that in the Map, you delete the name from the map instead.

Reply:
I decided to add `unmockTemplate(…)` to do that.